app/client/components/RegionFocusSwitcher.ts

```diff
@@ -294,19 +237,28 @@ export class RegionFocusSwitcher extends Disposable {
     }
     const comesFromKeyboard = initiator?.type === 'cycle';
     const panelElement = getPanelElement(current.id);
+    if (!panelElement) {
+      return;
+    }
+
+    // …Reset region focus switch if already on the panel itself
+    if (document.activeElement === panelElement) {
```

Comment:
Comment with very little context of the situation: `document.activeElement` doesn’t work well with Shadow DOM. I spent many hours in a11y-dialog tracking down bugs that were related to my usage of `document.activeElement` in contexts where Shadow DOM can exist. Just sharing that in case that’s a potential concern for you.

Reply:
Ah no, I meant custom elements and web components specifically (native DOM elements are totally fine).

This was an absolute PITA on a11y-dialog, and I’m thankful @mxmason helped with the groundwork. I’ve unfortunately learnt more about web components than I ever wanted to, and my takeaway is that it’s quite terrible to work with, and incredibly complicated (probably rightfully so, I don’t know).

Anyway, I’m happy to support with this. If you have any question, reach out or ping me and I’ll see if I can shed some light. :)

Reply:
Okay :) We don't use any custom elements here so I'll allow myself to just: ignore this. Until we need to stop ignoring it haha. Thanks for your help :handshake: